src/noisepy/seis/application_modules/esyn_utils.py

```diff
@@ -81,22 +98,30 @@ def ESYN_RadiaTrans_onesta(mean_free: float, tm: float, r: float, c: float) -> T
         Esyn = second
 
         return Esyn
+    if s0 <= 0:
```

Comment:
```suggestion
    else:
```
You could also have this check at the beginning of the function as validation and then not have the check for `s0>0` since at that point you know that's the case. A structure like:

```
def f(x):
   if not valid(x):
       raise ValueError(f"Invalid x: {x}")
   
   return some calculation with x 
```

Reply:
Sounds good. I will update it with this function.